src/main/java/com/google/devtools/build/lib/exec/SpawnRunner.java

```diff
@@ -198,6 +199,14 @@ default ArtifactPathResolver getPathResolver() {
 
     /** Reports a progress update to the Spawn strategy. */
     void report(ProgressStatus state, String name);
+
+    /**
+     * Returns a {@link MetadataInjector} that allows a caller to inject metadata about spawn
+     * outputs that are stored remotely.
+     */
+    default MetadataInjector getMetadataInjector() {
```

Comment:
This is also a risk for production code.

Reply:
done